aicsimageio/writers/rgb_writer.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from pathlib import Path
+from typing import Union
+
+import dask.array as da
+import numpy as np
+from imageio import imwrite
+
+from ..exceptions import InconsistentShapeError, InvalidDimensionOrderingError
+from ..transforms import reshape_data
+from .writer import Writer
+
+###############################################################################
+
+GREYSCALE_ORDER = "YX"
+RGB_ORDER = "YXC"
+
+###############################################################################
+
+
+class RGBWriter(Writer):
+    """
+    A writer for image data where the channel dimension act's as color. (RGB / RGBA)
+    Primarily directed at formats: "png", "jpg", etc.
+
+    This is primarily a passthrough to imageio.imwrite.
+    """
+
+    @staticmethod
+    def save(
+        data: Union[da.Array, np.ndarray],
+        filepath: Union[str, Path],
+        dim_order: str = None,
+        **kwargs,
+    ):
+        """
+        Write a data array to a file.
+
+        Parameters
+        ----------
+        data: Union[da.Array, np.ndarray]
+            The array of data to store. Data must have either two or three dimensions.
+        filepath: Union[str, Path]
+            The path to save the data at.
+        dim_order: str
+            The dimension order of the provided data.
+            Default: None. Based off the number of dimensions, will assume
+            the dimensions similar to how aicsimageio.readers.DefaultReader reads in
+            data. That is, two dimensions: YX and three dimensions: YXC.
+
+        Examples
+        --------
+        Data is the correct shape and dimension order
+
+        >>> image = dask.array.random.random((100, 100, 4))
+        ... RGBWriter.save(image, "file.png")
+
+        Data provided with current dimension order
+
+        >>> image = numpy.random.rand(3, 1024, 2048)
+        ... RGBWriter.save(image, "file.png", "CYX")
+        """
+        # Check filepath
+        # Do not fully resolve to absolute because Mac is bad with mounted drives
+        filepath = Path(filepath).expanduser()
+
+        # Assumption: if provided a dask array to save, it can fit into memory
+        if isinstance(data, da.core.Array):
+            data = data.compute()
+
+        # Check num dimensions
+        if len(data.shape) not in [2, 3]:
```

Comment:
can you write 4d gifs with this code?

Reply:
Yea the plan in my head is to add a `TimeseriesWriter` to utilize [`imageio.core.format.Writer.append_data`](https://imageio.readthedocs.io/en/stable/userapi.html?highlight=writer#imageio.core.format.Writer.append_data). Which would allow us to serialize large dask arrays with some compression.